### PR TITLE
fix: LSDV-5256: Stop automatically scrolling if current Paragraph is not on the screen

### PR DESCRIPTION
**Pull request by juliosgarbi**
### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [x] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)
_(check all that apply)_
- [x] Product design
- [x] Frontend

#### What is the new behavior?
If user scrolls the container and the current playing Paragraph is outside the screen the automatically scroll will stop work and it will start working back if user returns the focus to the current playing Paragraph

#### What is the current behavior?
If the automatically scroll is turned on, the Paragraphs will scroll to the current paragraph every time its change, doesn't matter if the current Paragraph is in focus or not

#### What libraries were added/updated?
no

#### Does this change affect performance?
no

#### Does this change affect security?
no

#### What alternative approaches were there?
NA

#### What feature flags were used to cover this change?
fflag_feat_front_lsdv_e_278_continuous_scrolling_of_paragraph_segments_with_audio

### Does this PR introduce a breaking change?
_(check only one)_
- [x] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [ ] No
- [ ] Not sure (briefly explain the situation below)

### What level of testing was included in the change?
_(check all that apply)_
- [x] e2e
- [ ] integration
- [ ] unit

### Which logical domain(s) does this change affect?
_(for bug fixes/features, be as precise as possible. ex. Authentication, Annotation History, Review Stream etc.)_

### CODE REVIEW

File: src/tags/object/Paragraphs/HtxParagraphs.js [hunk 1]
```diff
@@ -471,7 +489,16 @@ class HtxParagraphsView extends Component {
   }
 
   _handleScrollRoot() {
-    this._disposeTimeout();
+    if (this.activeRef.current) {
+      const { top, bottom, height } = this.activeRef.current.getBoundingClientRect();
+      const { offsetHeight, offsetTop } = this.myRef.current;
+      const offset = offsetTop + 95;
```
Comment (bmartel) [inline src/tags/object/Paragraphs/HtxParagraphs.js:495]: What is 95 here? If this is a constant we should store it as such at module level with a descriptive enough name.

**Review by bmartel — CHANGES_REQUESTED** [review]

File: src/tags/object/Paragraphs/HtxParagraphs.js (hunk 1 above)
Comment (yyassi-heartex) [inline src/tags/object/Paragraphs/HtxParagraphs.js:495]: what is 95?

File: src/tags/object/Paragraphs/HtxParagraphs.js [hunk 2]
```diff
@@ -427,32 +428,51 @@ class HtxParagraphsView extends Component {
...
       }
 
       this.lastPlayingId = item.playingId;
     }
   }
 
+  _handleScrollToPhrase() {
+    this.setState( { inViewPort : true });
+
+    const _padding = parseInt(window.getComputedStyle(this.myRef.current)?.getPropertyValue('padding-top')) || 0; // 8 is the padding between the phrases, so it will keep aligned with the top of the phrase
```
Comment (yyassi-heartex) [inline src/tags/object/Paragraphs/HtxParagraphs.js:467]: neat solution, i wasn't aware of `getComputedStyle` - very nice 👍

File: src/tags/object/Paragraphs/HtxParagraphs.js [hunk 3]
```diff
@@ -427,32 +428,51 @@ class HtxParagraphsView extends Component {
       const _wrapperOffsetTop = this.activeRef.current?.offsetTop - _padding;
       const _splittedText = 10; // it will be from 0 to 100% of the text height, going 10% by 10%
 
+      console.log('padding', _padding);
```
Comment (yyassi-heartex) [inline src/tags/object/Paragraphs/HtxParagraphs.js:431]: dont need that
```suggestion
```

File: src/tags/object/Paragraphs/HtxParagraphs.js [hunk 4]
```diff
@@ -427,32 +428,51 @@ class HtxParagraphsView extends Component {
...
+
```
Comment (yyassi-heartex) [inline src/tags/object/Paragraphs/HtxParagraphs.js:432]:
```suggestion
```

**Review by bmartel — APPROVED** [review]

File: src/tags/object/Paragraphs/Phrases.js [hunk 5]
```diff
@@ -20,9 +21,111 @@ const formatTime = (seconds) => {
...
+
+
+  // when user seek audio, the useEffect will be triggered and animate the reading line to the seek position
+  useEffect(() => {
+    setSeekAnimation(true);
+  }, [seek]);
+
+  // when user seek audio to a differente playing phrase, the useEffect will be triggered and animate the reading line to the seek position
+  useEffect(() => {
+    if (!isSeek) return;
```
Comment (bmartel) [inline src/tags/object/Paragraphs/Phrases.js:117]: This value will be stale referenced based on the last value of playingId, is that the intention or should isSeek be added to dependencies list?

Comment (juliosgarbi) [reply src/tags/object/Paragraphs/Phrases.js:117]: That is the intention, if user seeks to a different playingId, the invisible bar should start from the place that user sought, so here when playingId changes I'm just checking if user sought to this, and if user sought, the animation will start from the correct place to be sync

**Review by bmartel — APPROVED** [review]

**Review by yyassi-heartex — APPROVED** [review]

File: src/tags/object/Paragraphs/Phrases.js [hunk 6]
```diff
@@ -68,7 +175,12 @@ export const Phrases = observer(({ item, playingId, activeRef }) => {
           <span className={cls?.name} data-skip-node="true" style={style?.name}>{v[item.namekey]}</span>
         )}
 
-        <span className={cls?.text}>{v[item.textkey]}</span>
+        <span className={`${cls?.text} ${isFF(FF_LSDV_E_278) && styles.text}`}>
+          {v[item.textkey]}
+          {(isFF(FF_LSDV_E_278) && isActive) && (
+            <span ref={readingLineRef} className={`${styles.readingLine} reading-line`}></span>
+          )}
```
Comment (bmartel) [inline src/tags/object/Paragraphs/Phrases.js:182]:
```tsx
          {(isFF(FF_LSDV_E_278) && isActive) && (
            <span ref={readingLineRef} className={`${styles.readingLine} reading-line`}></span>
          )}
```
This is being added within the text content which the Paragraph parser is going to be influenced by, and likely is what was causing errors.

We have to make sure it is not counted within the paragraph data by either moving out of the span that also holds the text segment, or adding `data-skip-node="true"` to the reading line span.

**Review by bmartel — APPROVED** [review]

**Pull request merged**